src/commonMain/kotlin/com.adamratzman.spotify/SpotifyApi.kt

```diff
@@ -51,7 +55,7 @@ internal const val base = "https://api.spotify.com/v1"
  * @property logger The Spotify event logger
  *
  */
-abstract class SpotifyApi internal constructor(
+sealed class SpotifyApi<T : SpotifyApi<T, B>, B : ISpotifyApiBuilder<T, B>>(
```

Comment:
I really do not like having these unbounded type parameters

Reply:
Actually I could remove it again. My intention to add it, was that ISpotifyApiBuilder hadn't any `build` function. But I added them 😄.
Now it's just convenience that the `SpotifyAppApi` correlates to the `SpotifyAppApiBuilder` and the `SpotifyClientApi` to the `SpotifyClientApiBuilder` ([here](https://github.com/adamint/spotify-web-api-kotlin/pull/152/files#diff-fad0f6998edc6ed4c500548de7783016R132)). But I don't think it hurts that much, as it's just library internal. The `SpotifyApi` shouldn't be used from the outside.